src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala

```diff
@@ -697,60 +719,75 @@ trait V2Formats {
       maybeJson.foldLeft(appJson)((result, obj) => result ++ obj)
     }
 
-  implicit lazy val V2AppUpdateReads: Reads[AppUpdate] = {
-
-    (
-      (__ \ "id").readNullable[PathId].filterNot(_.exists(_.isRoot)) ~
-      (__ \ "cmd").readNullable[String](Reads.minLength(1)) ~
-      (__ \ "args").readNullable[Seq[String]] ~
-      (__ \ "user").readNullable[String] ~
-      (__ \ "env").readNullable[Map[String, String]] ~
-      (__ \ "instances").readNullable[Integer](minValue(0)) ~
-      (__ \ "cpus").readNullable[JDouble](greaterThan(0.0)) ~
-      (__ \ "mem").readNullable[JDouble] ~
-      (__ \ "disk").readNullable[JDouble] ~
-      (__ \ "executor").readNullable[String](Reads.pattern("^(//cmd)|(/?[^/]+(/[^/]+)*)|$".r)) ~
-      (__ \ "constraints").readNullable[Set[Constraint]] ~
-      (__ \ "uris").readNullable[Seq[String]] ~
-      (__ \ "storeUrls").readNullable[Seq[String]] ~
-      (__ \ "ports").readNullable[Seq[Integer]](uniquePorts) ~
-      (__ \ "requirePorts").readNullable[Boolean] ~
-      (__ \ "backoffSeconds").readNullable[Long].map(_.map(_.seconds)) ~
-      (__ \ "backoffFactor").readNullable[JDouble] ~
-      (__ \ "maxLaunchDelaySeconds").readNullable[Long].map(_.map(_.seconds)) ~
-      (__ \ "container").readNullable[Container] ~
-      (__ \ "healthChecks").readNullable[Set[HealthCheck]] ~
-      (__ \ "dependencies").readNullable[Set[PathId]]
-    )(AppUpdate(_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _)).flatMap { update =>
-        // necessary because of case class limitations (good for another 21 fields)
-        case class ExtraFields(
-          upgradeStrategy: Option[UpgradeStrategy],
-          labels: Option[Map[String, String]],
-          version: Option[Timestamp],
-          acceptedResourceRoles: Option[Set[String]],
-          ipAddress: Option[IpAddress])
-
-        val extraReads: Reads[ExtraFields] =
-          (
-            (__ \ "upgradeStrategy").readNullable[UpgradeStrategy] ~
-            (__ \ "labels").readNullable[Map[String, String]] ~
-            (__ \ "version").readNullable[Timestamp] ~
-            (__ \ "acceptedResourceRoles").readNullable[Set[String]](nonEmpty) ~
-            (__ \ "ipAddress").readNullable[IpAddress]
-          )(ExtraFields)
-
-        extraReads.map { extraFields =>
-          update.copy(
-            upgradeStrategy = extraFields.upgradeStrategy,
-            labels = extraFields.labels,
-            version = extraFields.version,
-            acceptedResourceRoles = extraFields.acceptedResourceRoles,
-            ipAddress = extraFields.ipAddress
-          )
-        }
+  private[this] def addUrisToFetch(fetch: Option[Seq[FetchUri]], uris: Option[Seq[String]]): Option[Seq[FetchUri]] = {
+    if (uris.nonEmpty) {
+      val converted = uris.getOrElse(Seq.empty).map(uri => FetchUri(uri = uri, extract = FetchUri.isExtract(uri)))
```

Comment:
probably misleading comment - sorry.
I don't think you have to merge here - correct?
Either map defined `uris` or use `fetch` as is.